File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        r = len(numbers)-1
        l = 0
        ans = []
        while l <= r:
            if numbers[l] + numbers[r] == target:
                ans.append(l+1)
                ans.append(r+1)
                break;
            elif numbers[l]+numbers[r] < target:
                l = l + 1
            else:
                r = r - 1

        return ans

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>>ans;
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<int>cur;
        backTrack(0, cur, target, nums);
        return ans;
    }

    void backTrack(int i, vector<int>cur, int target, vector<int>nums) {
        if(target == 0) {
            ans.push_back(cur);
            return;
        }
        if(target< 0|| i >= nums.size()) {
            return;
        }
        cur.push_back(nums[i]);
        backTrack(i, cur, target-nums[i], nums);
        cur.pop_back();
        backTrack(i+1, cur, target, nums);
    }
};
